### DISCUSSION TITLE
create-next-app (v12.1.6)

### DISCUSSION THREAD

**Discussion by zagnut007** [vercel/next.js#37079] [Help]
When running `yarn create next-app` as of May 20, 2022, it looks like it's installing nextjs v12.1.6 and PostCSS is broken out of the box?

node v16.14.0
next v12.1.6

This is my output:
```
% yarn dev
yarn run v1.17.3
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
wait  - compiling...
event - compiled client and server successfully in 433 ms (119 modules)

wait  - compiling / (client and server)...
wait  - compiling...
error - ./node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[3].oneOf[4].use[1]!./node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[3].oneOf[4].use[2]!./styles/Home.module.css
TypeError: Cannot read properties of undefined (reading 'config')
(node:45399) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
(Use `node --trace-deprecation ...` to show where the warning was created)
wait  - compiling /_error (client and server)...
wait  - compiling...
error - ./node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[3].oneOf[4].use[1]!./node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[3].oneOf[4].use[2]!./styles/Home.module.css
TypeError: Cannot read properties of undefined (reading 'config')
error - Error: Cannot find module '/Users/kevinblank/Documents/elevate/elm-street-big-branch/.next/server/pages/index.js'
```

To solve this, I added PostCSS as separate modules: `yarn add postcss-flexbugs-fixes postcss-preset-env` and dropped a default `postcss.config.json` file in from Next's docs.

**Comment by icyJoseph** [comment]
Could you share more about your project? `next.config.js`? What is it your are doing? I just made a project with `yarn create next-app`, and it starts normally.

**Reply by zagnut007** [reply]
There are not files created except for the defaults generated. Strangely I tried it on another machine without issue. When I looked at my global npm packages, nothing seemed different between the two machines. One machine is running MacOS 12.4 (the successful version) and one machine is running MacOS 11.6.5 (the one with issues) so maybe there's another package/software causing the issue? Xcode? I tried updating npm and yarn to see if that made any difference and it didn't.

**Reply by icyJoseph** [reply]
Oh wow that's shady...

Perhaps, if you create a `.babelrc` file at the root of the project?

```shell
{
  "presets": ["next/babel"],
  "plugins": []
}
```

That way we can discard that SWC is the problem.

**Reply by zagnut007** [reply]
Here's what I'm seeing after adding in the .babelrc, deleting the node_modules folder and starting fresh:

% yarn install
yarn install v1.22.18
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "eslint-config-next > @typescript-eslint/parser > @typescript-eslint/typescript-estree > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
[4/4] 🔨  Building fresh packages...
✨  Done in 12.84s.
 % yarn dev
yarn run v1.22.18
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
wait  - compiling...
error - ./node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[3].oneOf[8].use[1]!./node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[3].oneOf[8].use[2]!./styles/globals.css
TypeError: Cannot read properties of undefined (reading 'config')

**Reply by icyJoseph** [reply]
Interesting, I found this issue at the postcss-loader github page, not helpful, but it seems like a similar problem: https://github.com/webpack-contrib/postcss-loader/issues/584, it just ends up with a different claim, `undefined.config`

Too bad I can't reproduce this. Could you perhaps put that repo on Github? With its yarn.lock file?

**Comment by olhacodes** [comment]
I resolved the same issue yesterday by installing `"next-transpile-modules": "^9.*"` and `"@babel/preset-env": "^7.22.20"`. 
Also, make sure your next.configs are using withTM plugin.

`next.config.js`

```
const withTM = require('next-transpile-modules')([
  'your-modules',
]);

module.exports = withTM({});`
```

`.babelrc`

```
{
  "presets": ["next/babel"],
  "plugins": []
}
```

`package.json`

```
"dependencies": {
    "next": "^12.3.4",
    "next-transpile-modules": "^9.*",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
  },
```
